packages/2d/src/lib/components/RiveAnimation.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  BBox,
+  SignalValue,
+  SimpleSignal,
+  useThread,
+  viaProxy,
+} from '@revideo/core';
+import Rive, {
+  Artboard,
+  File,
+  LinearAnimationInstance,
+  Renderer,
+  RiveCanvas,
+} from '@rive-app/canvas-advanced';
+import {computed, initial, nodeName, signal} from '../decorators';
+import {Asset} from './Asset';
+import {RectProps} from './Rect';
+
+export interface RiveAnimationProps extends RectProps {
+  src?: SignalValue<string>;
+  artboardId?: SignalValue<string | number>;
+  animationId?: SignalValue<string | number>;
+}
+
+interface RiveAnimationInstance {
+  rive: RiveCanvas;
+  renderer: Renderer;
+  artboard: Artboard;
+  animation: LinearAnimationInstance;
+  canvas: HTMLCanvasElement;
+}
+
+@nodeName('RiveAnimation')
+export class RiveAnimation extends Asset {
+  @initial(0)
+  @signal()
+  public declare readonly artboardId: SimpleSignal<number | string, this>;
+
+  @initial(0)
+  @signal()
+  public declare readonly animationId: SimpleSignal<number | string, this>;
+
+  @initial(0)
+  @signal()
+  protected declare readonly time: SimpleSignal<number, this>;
+
+  protected currentTime: number = 0;
+  protected lastTime: number = 0;
+
+  public constructor(props: RiveAnimationProps) {
+    super(props);
+
+    const time = useThread().time;
+    const start = time();
+    this.time(() => time() - start);
+  }
+
+  @computed()
+  private async rive(): Promise<RiveAnimationInstance> {
+    const src = viaProxy(this.fullSource());
+    const rive = await Rive({
+      locateFile: () =>
+        `https://unpkg.com/@rive-app/canvas-advanced@2.7.3/rive.wasm`,
```

Comment:
I think this should be shipped and served by us. You can check out how we do this for mp4.wasm inside of `vite-plugin/partials/wasmExporter.ts`